packages/authentication/src/__tests__/fixtures/services/basic-auth-user-service.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/authentication
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {inject} from '@loopback/context';
+import {HttpErrors} from '@loopback/rest';
+import {UserService} from '../../../services/user.service';
+import {UserProfile} from '../../../types';
+import {USER_REPO} from '../keys';
+import {BasicAuthenticationStrategyCredentials} from '../strategies/basic-strategy';
+import {User} from '../users/user';
+import {UserRepository} from '../users/user.repository';
+
+export class BasicAuthenticationUserService
+  implements UserService<User, BasicAuthenticationStrategyCredentials> {
+  constructor(
+    @inject(USER_REPO)
+    private userRepository: UserRepository,
+  ) {}
+
+  async verifyCredentials(
+    credentials: BasicAuthenticationStrategyCredentials,
+  ): Promise<User> {
+    if (!credentials) {
+      throw new HttpErrors.Unauthorized(`'credentials' is null`);
+    }
+
+    if (!credentials.email) {
```

Comment:
Basic auth allows `username`. Do we want generalize it as `username` instead of `email`?

Reply:
can't people log in with email and password?

Reply:
email can be a person's userid can it not?

Reply:
I can have two gmail addresses associated with my name and surname. So an email can serve as a username.

Reply:
I am trying to make use of
```ts
export interface UserProfile {
  id: string;
  name?: string;
  email?: string;
}
```
and in our user service interface, we have 
```ts
convertToUserProfile(user: U): UserProfile;
```

where we have a `user` object (we obtained by searching a database with user credentials) and then we convert the larger `user` object into a slimmer `user profile` we place on the context.

Why would we have the email field in UserProfile if we cannot use it?
@jannyHou , what are your thoughts? thx

Reply:
Not sure how relevant it is, just wanna point out that there is another discussion thread talking about making the UserProfile more flexible: https://github.com/strongloop/loopback-next/issues/2246.

Reply:
I'm proposing to be consistent with http://www.passportjs.org/docs/basic-digest/. Please `email` id can be used as `username`.